### PR TITLE
feat: adds support for proxy-wasm-sdk.

### LINKED ISSUE #210: Tinygo support
## Summary

Tinygo is desirable as it would open doors for delivering coraza features in embedded runtimes like wasm. Currently tinygo has a set of limitations in terms of support of stdlib and hence we need to decide if we want to support tinygo as a first class or shall we be able to deliver tinygo integration (which requires coraza to be more flexible)

Notably, libraries like `encoding/xml` isn't supported and there is no easy replacement for it.

### Motivation

Provide coraza as a wasm plugin for envoy.

### PR DESCRIPTION
**Pull request by jcchavezs**
Closes #210

- [x] My code includes positive and negative tests.
- [x] I have an appropriate description with correct grammar.
- [x] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v2sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).

### CODE REVIEW

**Comment by jptosso** [comment]
Other pending tasks for this pr:

- Update README with tinygo instructions. We must specify that CRS tests won't work as they require XML body processing
- Add github actions to test tinygo compilation and regression tests

File: examples/wasm/Makefile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+build:
+	tinygo build -o main.wasm -scheduler=asyncify -target=wasi ./main.go
```
Comment (jcchavezs) [inline examples/wasm/Makefile:2]:
```suggestion
	tinygo build -o main.wasm -scheduler=none -target=wasi ./main.go
```

File: go.mod [hunk 2]
```diff
@@ -1,11 +1,18 @@
 module github.com/corazawaf/coraza/v2
 
-go 1.16
+go 1.17
```
Comment (jcchavezs) [inline go.mod:3]: Can be reverted.

**Comment by jptosso** [comment]
should we close this pr?

**Comment by jcchavezs** [comment]
I think so. @anuraaga had covered most (if not all) the problems in the
v3/dev branch.

**Pull request closed without merging**